packages/astro/src/core/app/node.ts

```diff
@@ -151,3 +168,36 @@ export async function loadApp(rootFolder: URL): Promise<NodeApp> {
 	const manifest = await loadManifest(rootFolder);
 	return new NodeApp(manifest);
 }
+
+/**
+ * Takes in a nullable WebAPI Headers object and produces a NodeJS OutgoingHttpHeaders object suitable for usage
+ * with ServerResponse.writeHead(..) or ServerResponse.setHeader(..)
+ *
+ * @param webHeaders WebAPI Headers object
+ * @returns NodeJS OutgoingHttpHeaders object with multiple set-cookie handled as an array of values
+ */
+export const createOutgoingHttpHeaders = (
```

Comment:
Does it need to be exported? If so, can we add an example in the JSDoc to show how to use it?

Reply:
It needs to be exported for unit testing. However, it can be exported from another file that is not exposed from the astro package.

Reply:
Then maybe it's having this function inside the test utilities

Reply:
Moved away from public API in the latest commit.